features/ultils/credentials.ts

```diff
@@ -0,0 +1,151 @@
+import type { Credential, Permission } from "./types";
+import type { ErrorObject } from "ajv";
+import { KintoneRestAPIClient } from "@kintone/rest-api-client";
+import fs from "fs";
+import path from "path";
+import Ajv from "ajv";
+import * as core from "@actions/core";
+
+type Record = {
+  key: {
+    type: "SINGLE_LINE_TEXT";
+    value: string;
+  };
+  app_id: {
+    type: "SINGLE_LINE_TEXT";
+    value: string;
+  };
+  api_tokens: {
+    type: "SUBTABLE";
+    value: Array<{
+      id: string;
+      value: {
+        token: {
+          type: "SINGLE_LINE_TEXT";
+          value: string;
+        };
+        permissions: {
+          type: "CHECK_BOX";
+          value: Permission[];
+        };
+      };
+    }>;
+  };
+};
+
+const e2eCredentialFileName = ".e2e-credentials.json";
+const e2eCredentialFilePath = path.join(
+  __dirname,
+  `../../${e2eCredentialFileName}`,
+);
+
+const isRunOnActions = () => !!process.env.GITHUB_ACTIONS;
+
+export const loadCredentials: () => Promise<Credential[]> = async () => {
+  if (!isRunOnActions() && fs.existsSync(e2eCredentialFilePath)) {
+    return loadFromFile();
+  }
+
+  const credentials = await loadFromKintone();
+  if (isRunOnActions()) {
+    credentials.forEach((credential) => {
+      credential.apiTokens.forEach((apiToken) => {
+        if (apiToken.token) {
```

Comment:
Is it confirming that the token is not an empty string?
If so, please check the length of the string.

```suggestion
        if (apiToken.token.length > 0) {
```